packages/playwright-msw/src/utils.ts

```diff
@@ -37,12 +37,31 @@ export const getHandlerPath = (
   return (<RestHandler>handler).info.path;
 };
 
-export const convertMswPathToPlaywrightUrl = (path: Path): string | RegExp => {
+export const convertMswPathToPlaywrightUrl = (path: Path): RegExp => {
+  // If already a regex, just return straight away
   if (path instanceof RegExp) {
     return path;
   }
-  const transformedPath = path.replace(/\/:[^/]+/g, '/*').replace(/\?.+/, '');
-  return transformedPath.endsWith('*')
-    ? transformedPath
-    : `${transformedPath}*`;
+
+  const originRegex = /(\w+:\/\/[^/]+)/u;
+  return new RegExp(
+    [
+      // Anchor to start of string
+      '^',
+      // Add optional origin if path is a relative url
+      originRegex.test(path) ? '' : `${originRegex.source}?`,
+      // Mutate provided path
+      path
+        // Strip query parameters
+        .replace(/\?.*$/, '')
```

Comment:
I don't think we really need it because I didn't see that query parameters are specified directly on backend routing.

In MSW there is even a warning for that https://github.com/mswjs/msw/blob/main/src/handlers/RestHandler.ts#L149.

It's ok to leave it since it doesn't change anything but it's probably redundant.

Reply:
We can simply regex if we're going to remove query params handling here.

Reply:
Ah, this might seem redundant, but it's actually quite intentional. 

When matching playwright routes using Regex patterns, it'll include the query parameters. The regex pattern I'm generating adds support for optional query parameters. If I don't first strip any existing query parameters from the string, it's possible that the generated regex will attempt to match two sets of query params. The reason why this might happen is because even though MSW warns about query params, a consumer technically can still specify them - MSW will just ignore them.

Reply:
I even wrote some tests to make sure that this is actually required. This is what happens if we remove it:
![image](https://user-images.githubusercontent.com/22004537/204930470-a4d3a762-9d02-42e5-860a-e9f35a34f137.png)